src/com/mambu/apisdk/services/SearchService.java

```diff
@@ -121,21 +120,97 @@ public SearchService(MambuAPIService mambuAPIService) {
 			String offset, String limit) throws MambuApiException {
-		// Available since Mambu 3.12. See MBU-8986, MBU-8975
-		// For NotificationMessages available since Mambu 3.14. See MBU-10646
-		// Specifying the sort order is available since Mambu 3.14. See MBU-10444
-		// POST {JSONFilterConstraints} /api/savings/transactions/search?offset=0&limit=5
 
-		// Example:: POST /api/loans/search {
-		// "filterConstraints":[{"filterSelection":"CREATION_DATE", "filterElement":"BETWEEN", "value":"2000-01-01",
-		// "secondValue":"2072-01-01", "dataItemType":"CLIENT" }],
-		// "sortDetails":{"sortingColumn":"ACCOUNT_ID", "sortingOrder":"ASCENDING", "dataItemType":"LOANS"}}
+		return searchEntitiesWithFullDetails(searchEntityType, filterConstraints, offset, limit);
 
-		ApiDefinition apiDefinition = SearchService.makeApiDefinitionforSearchByFilter(searchEntityType);
+	}
+
+	/**
+	 * Convenience method to GET Mambu entities by specifying filter constraints. This generic method can be used to
+	 * retrieve entities by filter constraints for any supported entity type. API users can also use methods specific to
+	 * each entity, for example to get clients by filter constraints use
+	 * {@link ClientsService#getClients(JSONFilterConstraints, String, String)}
+	 *
+	 * Note: the retrieved entities will not contain custom fields. In case custom fields are needed
+	 * {@link SearchService#searchEntitiesWithFullDetails(MambuEntityType, JSONFilterConstraints, String, String)} must be used
+	 *
+	 * @param searchEntityType
+	 *            Mambu entity type. Must not be null. Currently searching with filter constrains API supports the
+	 *            following entities: Clients, Groups, Loans, Savings, Loan Transactions, SavingsTransactions and
+	 *            NotificationMessages
+	 * @param filterConstraints
+	 *            JSONFilterConstraints object defining an array of applicable filter constraints and an optional sort
+	 *            order. Must not be null
+	 * @param offset
+	 *            pagination offset. If not null it must be an integer greater or equal to zero
+	 * @param limit
+	 *            pagination limit. If not null it must be an integer greater than zero
+	 *
+	 * @return list of entities of the searchEntityType matching provided filter constraints
+	 * @throws MambuApiException in case exception occurs while fetching entities
+	 */
+	public <T> List<T> searchEntitiesWithBasicDetails(MambuEntityType searchEntityType, JSONFilterConstraints filterConstraints,
+									  String offset, String limit) throws MambuApiException {
+
+//		 Available since Mambu 3.12. See MBU-8986, MBU-8975
+//		 For NotificationMessages available since Mambu 3.14. See MBU-10646
+//		 Specifying the sort order is available since Mambu 3.14. See MBU-10444
+//		 POST {JSONFilterConstraints} /api/savings/transactions/search?offset=0&limit=5
+//
+//		 Example:: POST /api/loans/search {

```

Comment:
do we need double colon after Example? Same for the other method below.

Reply:
removed from both places, thanks!